bpf/lib/encap.h

```diff
@@ -139,15 +139,16 @@ __encap_with_nodeid(struct __ctx_buff *ctx, __u32 tunnel_endpoint,
 		return DROP_WRITE_ERROR;
 
 	send_trace_notify(ctx, TRACE_TO_OVERLAY, seclabel, 0, 0, ENCAP_IFINDEX,
-			  0, monitor);
+			  (enum trace_reason)ct_reason, monitor);
```

Comment:
Can the function parameter just take an `enum trace_reason`?

Reply:
I only commented on this case since I saw the cast here in the first patch, but I think there's a similar question to ask for the later patches as well.

Reply:
Makes sense, thanks for catching this. I've addressed it.